### PR TITLE
chore: add script to migrate action settings

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

in #822 we have include a future breaking change with a warning.
We have to change the shape of actions:

the following attributes was support to be expression because of their type (boolean):
* active
* inProgress
* disabled
* available

It's still supported but deprecated and so will be removed.

**What is the chosen solution to this problem?**

We are adding a script to help you to migrate your action settings
Remember you are supposed to call your action by ID so all theses props are supposed to be in settings.

To apply the migration:

`./scripts/migrate_actions_settings_expressions.js -p ../myproject/src/settings.json`

If you are using a folder with multiple json:
`./scripts/migrate_actions_settings_expressions.js -f ../myproject/src/settings`

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: scripts/migrate_actions_settings_expressions.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+program
+	.version(stack_version)
+	.option('-d, --debug', 'display more info')
+	.option('-q, --quiet', 'do not display logs')
+	.option('-p, --path [value]', 'path of the settings.json to update')
+	.option('-f, --folder [value]', 'path of the folder containain json to update');
+
+program.on('--help', () => {
+	console.log('To update your project settings : ');
+	console.log('>node bin/version.js --path ../yourapp/settings.json');
```
Comment (jare-talend) [inline scripts/migrate_actions_settings_expressions.js]: update the help with the name of the script migrate_actions_settings_expressions.js

File: scripts/migrate_actions_settings_expressions.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+if (program.debug) {
+	console.log(`use stack version ${stack_version}`);
+}
+
+
+const JSONs = [];
+
+function readFolderContent(folder) {
+	console.log('scan ', folder)
```
Comment (jsomsanith) [inline scripts/migrate_actions_settings_expressions.js]: [Style] Missing `;`

File: scripts/migrate_actions_settings_expressions.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ * from version XX to version YY we have introduce the following change:
+ * action.available = 'myExpression' was evaluated using cmf expression
+ * @param {Object} settings
+ */
+function getActions(settings) {
+	if (!settings || !settings.actions) {
+		return {};
+	}
+	const actions = {};
+	Object.keys(settings.actions).map(key => {
```
Comment (jsomsanith) [inline scripts/migrate_actions_settings_expressions.js]: `.map()` --> `.forEach()`, you don't map anything, but process something and insert in a new object

File: scripts/migrate_actions_settings_expressions.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+		const newAction = Object.assign({}, action);
+		ATTRS.forEach(attr => {
+			const value = action[attr];
+			if (typeof value === 'string' || typeof value === 'object') {
+				console.log('update ', attr);
+				newAction[`${attr}Expression`] = value;
+				delete newAction[attr];
+			}
+		});
+		if (typeof newAction['labelExpression'] === 'string') {
```
Comment (jsomsanith) [inline scripts/migrate_actions_settings_expressions.js]: [Style] use dot expression

File: scripts/migrate_actions_settings_expressions.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+			console.log('You should use object to set labelExpression');
+		}
+		actions[key] = newAction;
+	});
+	return actions;
+}
+
+function save(ppath, data) {
+	const withLineEnd = data + '\n';
```
Comment (jsomsanith) [inline scripts/migrate_actions_settings_expressions.js]: [Style] Use string template

Comment (jsomsanith) [reply scripts/migrate_actions_settings_expressions.js]: can you rename that `DataWithLineEnd` please ?

File: scripts/migrate_actions_settings_expressions.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+	if (!program.quiet) {
+		console.log(`save ${ppath}`);
+	}
+	fs.open(ppath, 'w', (err, fd) => {
+		if (err) {
+			throw new Error(`error opening file: ${err}`);
+		}
+
+		fs.write(fd, withLineEnd, 0, data.length, null, (err) => {
```
Comment (jsomsanith) [inline scripts/migrate_actions_settings_expressions.js]: You seem to mix `data` and `dataWithLineEnd`
`fs.write(fd, dataWithLineEnd, 0, dataWithLineEnd.length, null, cb)`

**Pull request merged**